### ISSUE TITLE
iOS: NSCameraUsageDescription key required even if not used

### ISSUE THREAD

**Issue by benomatis** [stripe/stripe-react-native#210]
Related to #167 

@thorsten-stripe I just ran into this as well, I'm being asked to provide a NSCameraUsageDescription key even though I'm not using the camera in my app. However, the message is not coming during a review, so I cannot contact the review team, it's a request for me to add the key, otherwise the build won't end up in App Store Connect at all. I think we should, at the minimum, update the doc to say this key will need to be added no matter what, which this message seems to indicate. Here is the full text:

> Dear Developer,
> 
> We identified one or more issues with a recent delivery for your app, "App Name" 1.2.3 (45). Please correct the following issues, then upload again.
> 
> ITMS-90683: Missing Purpose String in Info.plist - Your app's code references one or more APIs that access sensitive user data. The app's Info.plist file should contain a NSCameraUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. Starting Spring 2019, all apps submitted to the App Store that access user data are required to include a purpose string. If you're using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. You can contact the developer of the library or SDK and request they release a version of their code that doesn't contain the APIs. Learn more (https://developer.apple.com/documentation/uikit/core_app/protecting_the_user_s_privacy).
> 
> Best regards,
> 
> The App Store Team

**Comment by benomatis** [comment]
@thorsten-stripe as an additional question, closely related, is card scanning available at all in iOS (or Android, for that matter)? I see no reference to it in this library even though the key above is set and says "to scan cards" - I read about the feature in [the Stripe iOS SDK docs](https://stripe.dev/stripe-ios/docs/#card-scanning-beta).

**Comment by davidme-stripe** [comment]
Card scanning is built-in on iOS — if you set an NSCameraUsageDescription, it'll show a "Scan card" button in PaymentSheet.

This App Store Connect message seems overly-broad: We don't show the card scanning button unless an `NSCameraUsageDescription` is set, and this check makes it difficult to distribute an SDK with an optional camera feature. It looks like we'll have to split card scanning into a separate component or a compile-time option.

For now, I'd recommend setting the `NSCameraUsageDescription` to "To scan cards", and we'll look into a way to disable this at compile-time.

**Comment by benomatis** [comment]
I have only gotten this message once, when I started using this library, and I did not have the key set. I already set the key, so it's ok, but would be indeed nice if it was optional, though I saw nothing changing in the app, I thought it will ask for permission to use the camera, but it doesn't.

**Comment by davidme-stripe** [comment]
The "scan card" button is only in [PaymentSheet](https://stripe.com/docs/payments/accept-a-payment?platform=react-native) at the moment, so it won't appear if you built your own card form.

**Comment by benomatis** [comment]
oh, ok, I'm using the card form. Is the PaymentSheet the name of the pre-built UI?

**Comment by davidme-stripe** [comment]
Yes. Card scanning outside of PaymentSheet is on our roadmap, but we don't have an ETA yet.

**Comment by benomatis** [comment]
ok, thank you. So I guess this issue could be used to prevent having to set
this key if Camera is otherwise not used in the app?

**Comment by davidme-stripe** [comment]
Yeah, we'll leave this open to track that work.

**Comment by harbolaez** [comment]
I try adding the  “To scan cards” but still getting the message from apple preventing to release the build. Any tips on how to proceed? Should i contact apple?

**Comment by JuliusHuizing** [comment]
Perhaps make this requirement more explicit in the stripe docs for eg implementing apple pay on ios? As a newbie to Stripe, I believe I followed the apple pay implementation docs step by step but did not notice this as a requirement. 

In general, this is very unexpected to me and it took me a few hours to find the issue because I make no use of the camera and apple does not specify the package that triggers the error. Just happy I stumbled upon this thread relatively quickly…

**Comment by zyaanbookee** [comment]
If using expo, I resolved this issue by adding in app.json
`{
  "expo": {
    "ios": {
      "infoPlist": {
        "NSCameraUsageDescription": "This app uses camera to scan."
      },
    },
   },
}`
